mantidproject/mantid
# ISIS Powder - Migration of GEM

Description of work.
This PR adds the GEM instrument into the ISIS Powder scripts. 

It contains two .cal files which normally should be added as external data. However the users specifically requested I add an example of the file structure in the respective instrument examples folder. This means they can copy the example folder out, mimic the structure and copy the relevant files in on their machines.

**To test:**
Download the following .zip file which contains all files required for testing: https://uploadfiles.io/da5bb1
- Extract files
- There is a minor mistake in the mapping file I created for this PR. In `<extracted_folder>/Calibration/Gem_Mapping.yaml` the value `rietveld` needs to be changed to `Rietveld` (capital R). The script will correctly inform you of this mistake if you forget to make this change
- Add `<extracted_folder>/InputData` to the list of user directories in Mantid so it can find the data
- Within the extracted folder are two scripts - one for creating the calibration and one for focusing. In both of them make sure you set the `path_to_extracted_file` variable to your extracted folder path.
- Try running focus first - it should fail and prompt you to run the calibration
- Run calibration then focus, they should both succeed 
- Try messing around with the various kwargs. Try removing, misspelling and adding additional keyword args. It should handle them all appropriately and give sane error messages
- Check the data looks sane. The focused silicon run should look like this (dSpacing and zoomed in): 
![gem si all](https://cloud.githubusercontent.com/assets/1817032/23807364/5e7b8662-05bd-11e7-9f36-c1ac83338025.PNG) (notice the sharp peaks)
- Code review

As always any and all suggestions and feedback is very welcome during this process. 

Fixes #18643 .

*Does not need to be in the release notes. - ISIS Powder scripts are still in development*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
